TCAT/Model/PlaceResult.swift

```diff
@@ -12,65 +12,74 @@ import SwiftyJSON
 import CoreLocation
 
 class PlaceResult: Place, JSONDecodable, CoordinateAcceptor {
-    
+
     var detail: String
     var placeID: String
-    
+
     private let detailKey = "detail"
     private let placeIDKey = "placeID"
-    
+
     init(name: String, detail: String, placeID: String) {
         self.detail = detail
         self.placeID = placeID
-        
+
         super.init(name: name)
     }
-    
+
     required convenience init(json: JSON) throws {
         let name = json["structured_formatting"]["main_text"].stringValue
         let detail = json["structured_formatting"]["secondary_text"].stringValue
         let placeID = json["place_id"].stringValue
-        
+
         self.init(name: name, detail: detail, placeID: placeID)
     }
-    
+
     override func isEqual(_ object: Any?) -> Bool {
-        if (!super.isEqual(object)) {
+        if !super.isEqual(object) {
             return false
         }
-        
+
         guard let object = object as? PlaceResult else {
             return false
         }
-        
+
         return object.placeID == placeID
     }
-    
+
     // MARK: Print
-    
+
     override var description: String {
         return "PlaceResult(name: \(name), detail: \(detail), placeId: \(placeID))"
     }
-    
+
     // MARK: NSCoding
-    
+
     required init(coder aDecoder: NSCoder) {
-        detail = aDecoder.decodeObject(forKey: detailKey) as! String
-        placeID = aDecoder.decodeObject(forKey: placeIDKey) as! String
-        
+        if let detailKey = aDecoder.decodeObject(forKey: detailKey) as? String {
```

Comment:
You can use `??` here.

`placeID = (aDecoder.decodeObject(forKey: placeIDKey) as? String) ?? ""`